### DISCUSSION TITLE
Mutable list of strings instead enum for command argument type

### DISCUSSION THREAD

**Discussion by ryleu** [Minestom/Minestom#1124] [Help] [answered]
Is it possible to use a list of strings as the options for a command argument? I would use an enum, but the amount and value of the arguments can change.

My end goal is to create a `/instance set <inst>` command, where `inst` is from a list made from the registered instances. I don't want to just use the String type because I want to have auto complete.

**Comment by GoldenStack** [comment, marked as answer]
You can use `ArgumentString` and then use `Argument#setSuggestionCallback` to provide dynamic suggestions for the argument.

**Reply by ryleu** [reply]
Awesome, thank you!

**Reply by bananasmoothii** [reply]
Does it also work with ArgumentWord ?
Edit: yes it does

**Comment by Minikloon** [comment]
Another idea is that you can register every string as literals.
